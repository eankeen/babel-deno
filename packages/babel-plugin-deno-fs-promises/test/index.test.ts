import {
  getNodeFiles, readTranspileAndWrite, nodeRun, denoRun,
} from './test.utils';

const nodeFileNames: string[] = getNodeFiles();

// @ts-ignore nodeFileNames is being used before it is defined
for (const fileName of nodeFileNames) {
  test(`snapshot: transpiled ${fileName}.mjs matches ${fileName}.deno.js`, async () => {
    const {
      denoResult,
    } = await readTranspileAndWrite(fileName);

    expect(denoResult.code).toMatchSnapshot();
    expect(denoResult.ast).toMatchSnapshot();

    // expect((await nodeRun(fileName)).stdout).toBe(
    //   (await denoRun(fileName)).stdout,
    // );
  });
}
