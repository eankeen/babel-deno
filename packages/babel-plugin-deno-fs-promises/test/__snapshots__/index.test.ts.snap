// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: transpiled first.mjs matches first.deno.js 1`] = `
"(async () => {
  const contents = await Deno.readTextFile('./file');
  console.log(contents);
})();"
`;

exports[`snapshot: transpiled first.mjs matches first.deno.js 2`] = `null`;

exports[`snapshot: transpiled fs-promises-no-option.mjs matches fs-promises-no-option.deno.js 1`] = `
"(async () => {
  await Deno.chmod('chmod', mode);
  await Deno.chown('chown', uid, gid);
  await Deno.copyFile('copyFileSrc', 'copyFileDest');
  await Deno.copyFile('copyFileSrc2', 'copyFileDest2');
  await Deno.mkdir('mkdir');
  await Deno.mkdir('mkdir2', {
    recursive: false,
    mode: 0o744
  });
  await fs.promises.mkdtemp('mkdirtemp');
  await fs.promises.mkdtemp('mkdirtemp2', {
    encoding: 'utf8'
  });
  await fs.promises.open(path, flags);
})();"
`;

exports[`snapshot: transpiled fs-promises-no-option.mjs matches fs-promises-no-option.deno.js 2`] = `null`;
